Added functionality for editing tasks

totokartonio opened the pull request.

Commit:
Added functionality for editing tasks

iffjey commented on src/App.tsx:
```diff
@@ -41,6 +41,12 @@ function App() {
     setTasks((prevTasks) => prevTasks.filter((task) => task.id !== taskId));
   };
 
+  const handleTaskUpdate = (updatedTask: Task) => {
+    setTasks((prevTasks) =>
+      prevTasks.map((task) => (task.id === updatedTask.id ? updatedTask : task))
+    );
+  };
```
я бы сократил возможность до обновления только вэлью.

```suggestion
  const handleTaskUpdate = (id: string, value: string) => {
    ...
  };
```

iffjey replied:
я бы ещё унифицировал названия. У тебя есть `handleComplete` `handleTaskUpdate` `handleAddTask`
надо выбрать какой-то один способ написания типовых действий

iffjey commented on src/components/TaskItem/TaskItem.tsx:
```diff
@@ -1,19 +1,34 @@
-};
+const TaskItem = ({ task, onComplete, onDelete, onUpdate }: Props) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [taskText, setTaskText] = useState(task.title);
```
тоже про нэйминг, надо определиться это все-таки text или title

iffjey replied:
и если ты уже в контексте taskItem, необязательно всё называть с префиксом task

iffjey commented on src/components/TaskItem/TaskItem.tsx:
```diff
@@ -1,19 +1,34 @@
+    setIsEditing(true);
+  };
+
+  const handleChange = (value: string) => {
+    setTaskText(value);
+  };
```
можно и просто setTaskText передать, этот дополнительный хэндлер ничего не делает

iffjey commented on src/components/TaskItem/TaskItem.tsx:
```diff
@@ -1,19 +1,34 @@
+
+  const handleSave = () => {
+    setIsEditing(false);
+    onUpdate({ ...task, title: taskText });
```
в дополнение к идее про передачу только вэлью и айди
```suggestion
    onUpdate(task.id, taskText);
```

iffjey commented on src/components/TaskItem/TaskItem.tsx:
```diff
@@ -22,10 +37,26 @@ const TaskItem = ({ task, onComplete, onDelete }: Props) => {
-      <Button variant="secondary" size="small" onClick={handleEditTask}>
-        <EditIcon />
-      </Button>
+      {isEditing ? (
```
рабочий вариант, но я бы ещё подумал про то, что наверное во время editMode не особо имеет смысл показывать чекбокс и кнопку delete. возможно хочется чтобы вся таска приобретала вид рабочего поля, без лишних действий

iffjey commented on src/components/TasksList/TasksList.tsx:
```diff
@@ -2,21 +2,39 @@ import "./TasksList.css";
+  onDelete,
+  onUpdate,
+}: Props) => {
+  const [taskList, setTaskList] = useState(tasks);
```
вот это не понял

iffjey reviewed (approved):
хороший PR! есть пара комментов

Commit:
Correct edtitng

1. Editing now updates only title value;
2. useState was removed from TasksList;
3. Namings of handlers were corrected;
4. Namings inside TaskItem were corrected;
5. In edit mode checkbox and deleteButton are no longer dispalyed;
6. CSS for input was adjusted to make it take up the entire width of the task;
7. CSS for new-task-form was updated to align with the new input styling.
